### PR TITLE
Web auto park week4

### PR DESCRIPTION
**Pull request by dzmitri-litvinov**

### CODE REVIEW

File: WebAutopark/Controllers/VehicleController.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        }
+        public IActionResult Index(string order)
+        {
+            var vehicles = _vehicleRepository.GetAll();
+            vehicles = OrderVehicle(vehicles, order);
+
+            return View(vehicles);
+        }
+        
+        public IActionResult GetById(int id)
```
Comment (PushMeNow) [inline WebAutopark/Controllers/VehicleController.cs:31]: Usually this page has name Details.

File: WebAutopark/Controllers/VehicleController.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        {
+            Vehicle vehicle = _vehicleRepository.GetById(id);
+            VehicleType vehicleType = _vehicleTypeRepository.GetById(vehicle.VehicleTypeId);
+            vehicle.VehicleType = vehicleType;
+
+            ViewBag.TaxPerMonth = vehicle.GetTaxPerMonth().ToString("0.00");
+            ViewBag.MaxKm = vehicle.GetMaxKm().ToString("0.00");
```
Comment (PushMeNow) [inline WebAutopark/Controllers/VehicleController.cs:38]: Why do u need additional variables in viewbag? U can use auto-props in this case

File: WebAutopark/Controllers/VehicleController.cs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+            return View(vehicles);
+        }
+        
+        public IActionResult GetById(int id)
+        {
+            Vehicle vehicle = _vehicleRepository.GetById(id);
+            VehicleType vehicleType = _vehicleTypeRepository.GetById(vehicle.VehicleTypeId);
+            vehicle.VehicleType = vehicleType;
+
```
Comment (PushMeNow) [inline WebAutopark/Controllers/VehicleController.cs:36]: Wait what? U already have done it in vehicle repo.

File: WebAutopark/Controllers/VehicleController.cs [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        public IActionResult Delete(int id)
+        {
+            _vehicleRepository.Delete(id);
+            return RedirectToAction("Index");
+        }
+
+        private SelectList GetVehicleTypesSelectList()
+        {
+            var vehicleTypes = _vehicleTypeRepository.GetAll().ToList();
```
Comment (PushMeNow) [inline WebAutopark/Controllers/VehicleController.cs:80]: Hm... Why do you need cal ToList here?

File: WebAutopark/Controllers/VehicleController.cs [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+                    break;
+                case "mileageKmDesc":
+                    vehicles = vehicles.OrderByDescending(v => v.MileageKm);
+                    break;
+                default:
+                    break;
+            }
+
+            return vehicles;
+        }
```
Comment (PushMeNow) [inline WebAutopark/Controllers/VehicleController.cs:117]: Creat  new method in vehicle repo for ordering and make this action on sql side

File: WebAutopark/Extentions/VehicleExtentions.cs [hunk 6]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using WebAutopark.DAL.Entities;
+
+namespace WebAutopark.Extentions
+{
+    public static class VehicleExtentions
```
Comment (PushMeNow) [inline WebAutopark/Extentions/VehicleExtentions.cs:9]: I don't see the reason in these extensions. Do it by auto-props.

File: WebAutopark/Views/Vehicle/Create.cshtml [hunk 7]
```diff
@@ -0,0 +1,59 @@
+﻿@{
+    ViewData["Title"] = "Home Page";
```
Comment (PushMeNow) [inline WebAutopark/Views/Vehicle/Create.cshtml:2]: Why all your pages have title "Home page"?

**Review by PushMeNow — CHANGES_REQUESTED** [review]

File: WebAutopark/Controllers/VehicleController.cs (hunk 1 above)
Comment (dzmitri-litvinov) [reply WebAutopark/Controllers/VehicleController.cs:31]: image
Done

File: WebAutopark/Controllers/VehicleController.cs (hunk 2 above)
Comment (dzmitri-litvinov) [reply WebAutopark/Controllers/VehicleController.cs:38]: image
Done

File: WebAutopark/Controllers/VehicleController.cs (hunk 3 above)
Comment (dzmitri-litvinov) [reply WebAutopark/Controllers/VehicleController.cs:36]: image
Done

File: WebAutopark/Views/Vehicle/Create.cshtml (hunk 7 above)
Comment (dzmitri-litvinov) [reply WebAutopark/Views/Vehicle/Create.cshtml:2]: Fixed

File: WebAutopark/Extentions/VehicleExtentions.cs (hunk 6 above)
Comment (dzmitri-litvinov) [reply WebAutopark/Extentions/VehicleExtentions.cs:9]: image
Done

File: WebAutopark/Controllers/VehicleController.cs (hunk 4 above)
Comment (dzmitri-litvinov) [reply WebAutopark/Controllers/VehicleController.cs:80]: image
Removed

File: WebAutopark/Controllers/VehicleController.cs (hunk 5 above)
Comment (dzmitri-litvinov) [reply WebAutopark/Controllers/VehicleController.cs:117]: image
Done

File: WebAutopark.DAL/Repositories/VehicleRepository.cs [hunk 8]
```diff
@@ -35,18 +35,73 @@ public void Delete(int id)
...
+            }
+
+            switch (orderingDir)
+            {
+                case "desc":
+                    sqlOrdering += " DESC";
+                    break;
+                default:
+                    break;
+            }
```
Comment (UNDogMan) [inline WebAutopark.DAL/Repositories/VehicleRepository.cs:102]: U dont need switch case here. Use if.

File: WebAutopark/Controllers/OrderController.cs [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+        }
+
+        public IActionResult Details(int id)
+        {
+            var order = _orderRepository.GetById(id);
+            var vehicle = _vehicleRepository.GetById(order.VehicleId);
+            order.Vehicle = vehicle;
+            var ordersElements = _orderElementRepository.GetAll();
+
+            order.OrderElements = ordersElements.Where(o => o.OrderId == id).ToList();
```
Comment (UNDogMan) [inline WebAutopark/Controllers/OrderController.cs:39]: Move this to database as u did it for order-vehicle.

**Review by UNDogMan — CHANGES_REQUESTED** [review]

File: WebAutopark.DAL/Repositories/VehicleRepository.cs (hunk 8 above)
Comment (dzmitri-litvinov) [reply WebAutopark.DAL/Repositories/VehicleRepository.cs]: image
Done

File: WebAutopark/Controllers/OrderController.cs (hunk 9 above)
Comment (dzmitri-litvinov) [reply WebAutopark/Controllers/OrderController.cs:39]: image
It's done local and will be pushed in the next PR

**Review by UNDogMan — APPROVED** [review]

File: WebAutopark.DAL/Repositories/OrdersElementsRepository.cs [hunk 10]
```diff
@@ -13,7 +13,7 @@ public class OrdersElementsRepository : RepositoryBase, IRepository<OrderElement
     {
         private const string sqlQueryCreateString = "INSERT INTO OrdersElements (OrderId, SparePartId, SparePartQuantity) VALUES(@OrderId, @SparePartId, @SparePartQuantity)";
         private const string sqlQueryDeleteString = "DELETE FROM OrdersElements WHERE Id = @id";
-        private const string sqlQueryGetAllString = "SELECT * FROM OrdersElements";
+        private const string sqlQueryGetAllString = "SELECT OE.*, SP.Id SPID, SP.PartName FROM OrdersElements OE, SpareParts SP WHERE OE.SparePartId = SP.Id";
```
Comment (PushMeNow) [inline WebAutopark.DAL/Repositories/OrdersElementsRepository.cs:16]: It is bad syntax. Use JOIN here.

File: WebAutopark.DAL/Repositories/OrdersRepository.cs [hunk 11]
```diff
@@ -13,7 +13,9 @@ public class OrdersRepository : RepositoryBase, IRepository<Order>
     {
         private const string sqlQueryCreateString = "INSERT INTO Orders (VehicleId) VALUES(@VehicleId)";
         private const string sqlQueryDeleteString = "DELETE FROM Orders WHERE Id = @id";
-        private const string sqlQueryGetAllString = "SELECT * FROM Orders";
+        private const string sqlQueryGetAllString = "SELECT O.*, V.Id VID, V.VehicleTypeId, V.ModelName, V.RegistrationNumber, V.WeightKg, V.ManufactureYear, " +
+            "V.MileageKm, V.Color, V.Engine, V.EngineCapacity, V.Consumption, V.FuelTankOrBattery " +
+            "FROM Orders O, Vehicles V WHERE O.VehicleId = V.Id";
```
Comment (PushMeNow) [inline WebAutopark.DAL/Repositories/OrdersRepository.cs:18]: again use join

File: WebAutopark.DAL/Repositories/VehicleRepository.cs [hunk 12]
```diff
@@ -35,18 +35,69 @@ public void Delete(int id)
...
+        {
+            return connection.Query<Vehicle, VehicleType, Vehicle>(sqlQueryGetAllString + SqlOrdering(orderingCol, orderingDir),
+                (vehicle, vehicleType) =>
+                {
+                    vehicle.VehicleType = vehicleType;
+                    return vehicle;
+                }, splitOn: "VTID");
+        }
+
+        private string SqlOrdering(string orderingCol, string orderingDir)
```
Comment (PushMeNow) [inline WebAutopark.DAL/Repositories/VehicleRepository.cs:72]: Lets make 'orderingDir' like enum

File: WebAutopark.DAL/Repositories/VehicleRepository.cs [hunk 13]
```diff
@@ -35,18 +35,69 @@ public void Delete(int id)
...
+                    break;
+                default:
+                    orderingDir = "";
+                    break;
+            }
+
+            if (orderingDir == "desc")
+            {
+                sqlOrdering += " DESC";
+            }
```
Comment (PushMeNow) [inline WebAutopark.DAL/Repositories/VehicleRepository.cs:98]: Not need skip ASC ending. Make switch and add ASC or DESC. It will look more user friendly

**Review by PushMeNow — CHANGES_REQUESTED** [review]

File: WebAutopark.DAL/Repositories/VehicleRepository.cs (hunk 13 above)
Comment (dzmitri-litvinov) [reply WebAutopark.DAL/Repositories/VehicleRepository.cs:98]: image
done

File: WebAutopark.DAL/Repositories/VehicleRepository.cs (hunk 12 above)
Comment (dzmitri-litvinov) [reply WebAutopark.DAL/Repositories/VehicleRepository.cs:72]: image
Done

File: WebAutopark.DAL/Repositories/OrdersElementsRepository.cs (hunk 10 above)
Comment (dzmitri-litvinov) [reply WebAutopark.DAL/Repositories/OrdersElementsRepository.cs:16]: Will be fixed in the next PR

File: WebAutopark.DAL/Repositories/OrdersRepository.cs (hunk 11 above)
Comment (dzmitri-litvinov) [reply WebAutopark.DAL/Repositories/OrdersRepository.cs:18]: Will be fixed in the next PR

**Review by PushMeNow — APPROVED** [review]

**Pull request merged**